scylla-cdc/src/lib.rs

```diff
@@ -5,9 +5,233 @@ pub mod stream_generations;
+            .await?;
+        session
+            .query(
+                format!(
+                    "CREATE TABLE {} (pk {}, ck int, v int, primary key (pk, ck)) WITH cdc = {{'enabled' : true}}",
+                    table_name, pk_type_name
+                ),
+                (),
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    struct Test<'a, T: FromCqlVal<CqlValue> + Value + Eq + Hash + Send + Sync> {
+        session: Session,
+        performed_operations: HashMap<T, VecDeque<Operation>>,
+        table_name: &'a str,
+    }
+
+    impl<'a, T: 'static + FromCqlVal<CqlValue> + Value + Eq + Hash + Send + Sync> Test<'a, T> {
+        async fn new(
+            session: Session,
+            table_name: &'a str,
+            pk_type_name: &'a str,
+        ) -> anyhow::Result<Test<'a, T>> {
+            create_table_and_keyspace(&session, table_name, pk_type_name).await?;
+
+            Ok(Test {
+                session,
+                performed_operations: HashMap::new(),
+                table_name,
+            })
+        }
+
+        fn push_back(&mut self, pk: T, operation: Operation) {
+            self.performed_operations
+                .entry(pk)
+                .or_insert(VecDeque::new())
+                .push_back(operation);
+        }
+
+        async fn insert(&mut self, pk: T, ck: i32, v: Option<i32>) -> anyhow::Result<()> {
+            self.session
+                .query(
+                    format!(
+                        "INSERT INTO {} (pk, ck, v) VALUES (?, ?, ?)",
+                        self.table_name
+                    ),
+                    (&pk, ck, v),
+                )
+                .await?;
+            let operation = (OperationType::RowInsert, ck, v);
+            self.push_back(pk, operation);
+
+            Ok(())
+        }
+
+        async fn update(&mut self, pk: T, ck: i32, v: Option<i32>) -> anyhow::Result<()> {
+            self.session
+                .query(
+                    format!(
+                        "UPDATE {} SET v = ? WHERE pk = ? AND ck = ?",
+                        self.table_name
+                    ),
+                    (v, &pk, ck),
+                )
+                .await?;
+            let operation = (OperationType::RowUpdate, ck, v);
+            self.push_back(pk, operation);
+
+            Ok(())
+        }
+
+        async fn read_cdc(self) -> anyhow::Result<()> {
+            // [TODO]: CDC library is not fully implemented yet, this is a placeholder.
+            let factory = TestConsumerFactory::new(self.performed_operations);
+            let mut consumer = factory.new_consumer().await;
+
+            let mut rows_stream = self
+                .session
+                .query_iter(
+                    format!("SELECT * FROM {}_scylla_cdc_log", self.table_name),
+                    (),
+                )
+                .await?;
+            let first = rows_stream.next().await.unwrap()?;
```

Comment:
Why do you need to get the first row manually?

Reply:
To create the schema without complicating the loop body.
It's only a placeholder though, this function's body will be replaced as soon as the library is finished.

Reply:
Even if the code is going to be eventually replaced, I would like to know the reason. Why getting the first row is needed to create the schema? You already have the `rows_stream` object and the `get_column_specs()` method is supposed to work without before calling `next()` or iterating.

Reply:
Nevermind then, some kind of mind lag must have occured, you are correct.